frontend/src/routes/Governance/common/util.tsx

```diff
@@ -650,18 +655,108 @@ export function hasInformOnlyPolicies(items: Array<PolicyTableItem>): boolean {
   return true
 }
 
+// Transform string annotation to key and value pair type
+/* istanbul ignore next */
+export const transformAnnotation = (
+  anoString: string
+): {
+  [key: string]: string
+} => {
+  const parsed = anoString.split('; ')
+  const result: {
+    [key: string]: string
+  } = {}
+  for (const keyEqualVal of parsed) {
+    const kv = keyEqualVal.split('=')
+    if (kv.length == 2) {
+      result[kv[0]] = kv[1]
+    }
+  }
+
+  return result
+}
+
+// type guard
+function isPolicy(object: Policy | DiscoveredPolicyItem): object is Policy {
+  return object && (object as Policy).metadata !== undefined
+}
+
 export function getPolicySource(
-  policy: Policy,
+  policy: Policy | DiscoveredPolicyItem,
   helmReleases: HelmRelease[],
   channels: Channel[],
   subscriptions: Subscription[],
   t: TFunction<string, undefined>
 ): string | JSX.Element {
-  const isExternal = resolveExternalStatus(policy)
+  if (!policy) {
+    // Prevent error in detail page
+    return <></>
+  }
+
+  const isExternal = policy && isPolicy(policy) ? resolveExternalStatus(policy) : policy._isExternal || false
   let source: string | JSX.Element = t('Local')
   if (isExternal) {
-    const policySource = resolveSource(policy, helmReleases, channels, subscriptions)
+    const policySource = isPolicy(policy)
+      ? resolveSource(policy.metadata.annotations ?? {}, helmReleases, channels, subscriptions)
+      : resolveSource(transformAnnotation(policy.annotations), helmReleases, channels, subscriptions)
     source = policySource ? getSource(policySource, isExternal, t) : t('Managed externally')
   }
   return source
 }
+
+export function getEngineString(kind: string): string {
+  switch (kind) {
+    case 'ConfigurationPolicy':
+    case 'CertificatePolicy':
+    case 'OperatorPolicy':
+      return 'Open Cluster Management'
+    default:
+      return 'Unknown'
+  }
+}
+
+export function getEngineWithSvg(kind: string): JSX.Element {
+  switch (kind) {
+    case 'ConfigurationPolicy':
+    case 'CertificatePolicy':
+    case 'OperatorPolicy':
+      return (
+        <div style={{ display: 'flex', alignItems: 'center', gap: 8 }}>
+          <div
+            style={{
+              height: 18,
+              width: 18,
+            }}
+          >
+            {' '}
+            <OcmSvg />
+          </div>{' '}
+          Open Cluster Management
+        </div>
+      )
+    default:
+      return <>Unknown</>
+  }
+}
+
+// ex: The label looks like this: this "app=demo; open-cluster-management.io/policy-set=cluster-hardening",
+/* istanbul ignore next */
+export const getStringLabelValue = (labels: string, labelKey: string): string => {
```

Comment:
Could you reuse `transformAnnotation` here? If you can, please rename `transformAnnotation` to something more generic like `parseStringMap`.

Reply:
I tried but referring other functions in getStringLabelValue is not working sadly

Reply:
But I changed the name to `parseStringMap`